mappings/net/minecraft/advancement/criterion/ItemCriterion.mapping

```diff
@@ -14,19 +14,19 @@ CLASS net/minecraft/class_4711 net/minecraft/advancement/criterion/ItemCriterion
 			ARG 1 id
 			ARG 2 entity
 			ARG 3 location
-		METHOD method_27981 create (Lnet/minecraft/class_2090$class_2091;Lnet/minecraft/class_2073$class_2074;)Lnet/minecraft/class_4711$class_4712;
+		METHOD method_27981 createItemUsedOnBlock (Lnet/minecraft/class_2090$class_2091;Lnet/minecraft/class_2073$class_2074;)Lnet/minecraft/class_4711$class_4712;
 			ARG 0 location
 			ARG 1 item
 		METHOD method_43125 createAllayDropItemOnBlock (Lnet/minecraft/class_2090$class_2091;Lnet/minecraft/class_2073$class_2074;)Lnet/minecraft/class_4711$class_4712;
 			ARG 0 location
 			ARG 1 item
 		METHOD method_51709 create (Lnet/minecraft/class_2090$class_2091;Lnet/minecraft/class_2073$class_2074;Lnet/minecraft/class_2960;)Lnet/minecraft/class_4711$class_4712;
 			ARG 0 location
-			ARG 1 tool
+			ARG 1 item
 			ARG 2 id
 		METHOD method_51710 createPlacedBlock (Lnet/minecraft/class_2248;)Lnet/minecraft/class_4711$class_4712;
 			ARG 0 block
-		METHOD method_51711 testLocation (Lnet/minecraft/class_47;)Z
+		METHOD method_51711 test (Lnet/minecraft/class_47;)Z
```

Comment:
I remember this tests only the location LootContextPredicate, not the whole conditions

Reply:
The location *is* the whole conditions in an ItemCriterion - the entity one goes unused, untested, and unserialized.

Reply:
The location as far as I can see *is* the whole conditions in the case of an item criterion, and the entity one goes unused - location is all that's ever tested against, and all the create methods simply pass EMPTY for the entity.